mestopy/mestopy.py

```diff
@@ -1 +1,242 @@
+            inverted frequency responses. Data can be in domain 'freq' or
+            'time' and will be transformed if necessary.
+        sens : float, optional
+            Sensitivity of the device as a factor. If neither device_data nor
+            sens is given, add_device generates a device that has no effect to
+            the measurement chain as it has no frequency response and a
+            sesitivity (factor) of 1.
+        device_name : str, optional
+            The name of the new device to add to the masurement chain.
+            The default is an empty string.
+        """
+        if device_data is None:
+            device_data = Signal(np.full(int(self.sampling_rate/2), 1.0),
+                                 self.sampling_rate,
+                                 domain='freq')
+        # check if ref_signal is a pyfar.Signal, if not raise Error
+        if not isinstance(device_data, Signal):
+            raise TypeError('Input data must be of type: Signal.')
+        # check if there are no devices in measurement chain
+        if self.devices == []:
+            # add ref-measurement to chain
+            device_data.domain = 'freq'
+            new_device = DeviceObj(device_data,
+                                   sens,
+                                   device_name)
+            self.devices.append(new_device)
+        else:
+            # check if n_bins of all devices is the same
+            if not self.devices[0].data.n_bins == device_data.n_bins:
+                raise ValueError("ref_signal has wrong n_bins")
+            # check if sampling_rate of new device and MeasurementChain
+            # is the same
+            if not self.sampling_rate == device_data.sampling_rate:
+                raise ValueError("ref_signal has wrong samping_rate")
+            # add device to chain
+            new_device = DeviceObj(device_data,
+                                   sens,
+                                   device_name)
+            self.devices.append(new_device)
+
+    def list_devices(self):
+        """Returns a list of names of all devices in the measurement chain.
+        """
+        # list all ref-objects in chain
+        device_names = []
+        for dev in self.devices:
+            name = dev.device_name
+            device_names.append(name)
+        return device_names
+
+    def remove_device(self, num):
+        """Removes a single device from the measurement chain,
+         by name or number.
+
+        Attributes
+        ----------
+        num : int or str
+        Identifier for device to remove. Device can be found by name as string
+        or by number in device list as int.
+        """
+        # remove ref-object in chain position num
+        if isinstance(num, int):
+            front = self.devices[:num]
+            back = self.devices[num+1:]
+            for i in back:
+                front.append(i)
+            self.devices = front
```

Comment:
This should also do the trick :)
```suggestion
            self.devices.pop(num)
```